code/data_structures/src/queue/deque/deque.cpp

```diff
@@ -0,0 +1,211 @@
+#include <iostream>
+using namespace std;
+
+class DequeEmptyException
+{
+public:
+    DequeEmptyException()
+    {
+        cout << "Deque empty" << endl;
+    }
+};
+
+// Each node in a doubly linked list
+class Node
+{
+public:
+    int data;
+    Node* next;
+    Node* prev;
+};
+
+class Deque
+{
+private:
+    Node* front;
+    Node* rear;
+    int count;
+
+public:
+    Deque()
+    {
+        front = NULL;
+        rear = NULL;
+        count = 0;
+    }
+
+    void InsertFront(int element)
+    {
+        // Create a new node
+        Node* tmp = new Node();
+        tmp->data = element;
+        tmp->next = NULL;
+        tmp->prev = NULL;
+
+        if ( isEmpty() ) {
+            // Add the first element
+            front = rear = tmp;
+        }
+        else {
+            // Prepend to the list and fix links
+            tmp->next = front;
+            front->prev = tmp;
+            front = tmp;
+        }
+
+        count++;
+    }
+
+    int RemoveFront()
+    {
+        if ( isEmpty() ) {
```

Comment:
For if and loops that have one line, you should omit the braces. This is applicable for all the file.